urllib3/poolmanager.py

```diff
@@ -152,7 +202,11 @@ def urlopen(self, method, url, redirect=True, **kw):
         :class:`urllib3.connectionpool.ConnectionPool` can be chosen for it.
         """
         u = parse_url(url)
-        conn = self.connection_from_host(u.host, port=u.port, scheme=u.scheme)
+        request_context = self.connection_pool_kw.copy()
+        request_context['scheme'] = u.scheme or 'http'
+        request_context['port'] = u.port or port_by_scheme.get(u.scheme, 80)
+        request_context['host'] = u.host
```

Comment:
I am not crazy about how I changed ``connection_from_host`` to not really do what it says because it now takes all the ``connection_pool_kw`` into account. I meant to either change this back to ``connection_from_host`` and accept it being a bit odd or mark ``connection_from_host`` as deprecated (or something) and make something else, but clearly I forgot. Maybe this was a _bit_ too rough to be a PR, but I was worried that I was heading in the wrong direction entirely with the per-PoolManager keys.

In any case, this should change, but I'm not certain about how I want to do it yet.